lib/Routing/DynamicRouteProvider.php

```diff
@@ -68,6 +68,11 @@ public function addHandler(DynamicRouteHandlerInterface $handler)
     public function getRouteCollectionForRequest(Request $request): RouteCollection
     {
         $collection = new RouteCollection();
+
+        if($request->attributes->has('_controller')) {
```

Comment:
```suggestion
        if($request->get('_controller', false)) {
```
the param is in the query:
![image](https://user-images.githubusercontent.com/5137917/209659069-469c9377-458b-4a0f-9092-197d59049378.png)

Reply:
@dvesh3 For me it was in `attributes`, it's added to the request here: 
https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/HttpKernel/EventListener/FragmentListener.php#L73

Not sure if `$request->get()` could have some possible side-effects, as you can then use `?_controller=foo` to skip dynamic routing on any request 🤔

Reply:
It will be in the query for fragment endpoint (ESI), but that's a special case since the request is signed.

Reply:
@brusch the fragment request does not reach at line 73 because of this check here https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/HttpKernel/EventListener/FragmentListener.php#L61

Reply:
@dvesh3 the goal here is to avoid doing dynamic routing if Symfony's routing already sucessfully triggered (or wil

Reply:
@dvesh3 really? Why should that be set already, as it's in the `attributes` bag, that cannot be set by request params (e.g. `GET` ...). I think only the `_path` query param is set in the first place, right?

Reply:
Need to pay very close attention here because it's easy to introduce RCE here on accident.

Reply:
yes, this could lead to RCE so reverted the commit.